core/src/main/java/de/danoeh/antennapod/core/util/ShownotesLoader.java

```diff
@@ -0,0 +1,37 @@
+package de.danoeh.antennapod.core.util;
+
+import java.util.concurrent.Callable;
+
+import de.danoeh.antennapod.core.feed.FeedItem;
+import de.danoeh.antennapod.core.feed.FeedMedia;
+import de.danoeh.antennapod.core.storage.DBReader;
+
+public class ShownotesLoader {
+
+    public static Callable<String> loadShownotes(ShownotesProvider provider) {
+        if(provider instanceof FeedItem)
+            return loadForFeedItem((FeedItem) provider);
+        else if(provider instanceof FeedMedia)
```

Comment:
Hmm... This looks a bit ugly, to be honest. Pretty much fights the polymorphism. Is it actually needed to have a generic method that takes `ShownotesProvider`? Maybe it could just be two methods for `FeedMedia` and `FeedItem`.